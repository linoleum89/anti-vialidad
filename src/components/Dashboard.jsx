import React from "react";
import { Redirect } from "react-router-dom";
import { fetch_reports, fetch_sectors } from "../utils/utils";
import Report from "../components/Report";
import Loader from "../components/Loader";

class Dashboard extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      reports: []
    };
  }
  componentDidMount() {
    const { isValidLogin } = this.props;
    if (isValidLogin) {
      setTimeout(async () => {
        const reports = await fetch_reports();
        const sectors = await fetch_sectors();
        this.setState({
          reports: reports,
          sectors: sectors
        });
      }, 1000);
    }
  }
  render() {
    const { isValidLogin } = this.props;
    const reports = this.parseReports(this.state.reports, this.state.sectors);

    if (!isValidLogin) {
      return <Redirect to="/" />;
    }
    return reports && reports.length > 0 ? (
      reports
    ) : (
      <Loader>
        <div className="overlay">
          <h1>Loading reports...</h1>
        </div>
      </Loader>
    );
  }

  parseReports(reports = [], sectors = []) {
    return (
      reports &&
      reports.map(report => {
        const sector = sectors.find(sector => {
          return parseInt(sector.id) === report.sector;
        });

        report.className = (sector && sector.className) || "";

        return <Report key={report.name} {...report} />;
      })
    );
  }
}

export default Dashboard;
