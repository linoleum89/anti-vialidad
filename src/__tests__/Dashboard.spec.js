import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import Dashboard from "../components/Dashboard";
import { BrowserRouter as Router } from "react-router-dom";

describe("Dashboard Component", () => {
  let component, tree;
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
    component = TestRenderer.create(
      <Router>
        <Dashboard isValidLogin={true} />
      </Router>
    );
    tree = component.toJSON();    
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });
  it("should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
  it("should be a overlay loader before render the reports", () => {
    expect(tree.type).toBe("div");
    expect(tree.props.className).toBe("overlay");
  });
});
