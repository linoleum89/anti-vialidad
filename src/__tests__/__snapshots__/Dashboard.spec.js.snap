// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should match snapshot 1`] = `
<div
  className="overlay"
>
  <h1>
    Loading reports...
  </h1>
</div>
`;
